packages/functional-tests/tests/syncV3/signUpWithCWTS.spec.ts

```diff
@@ -0,0 +1,228 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { test, expect, newPagesForSync } from '../../lib/fixtures/standard';
+import uaStrings from '../../lib/ua-strings';
+import { FirefoxCommand, createCustomEventDetail } from '../../lib/channels';
+
+const password = 'passwordzxcv';
+let email;
+let syncBrowserPages;
+
+test.describe.configure({ mode: 'parallel' });
+
+test.describe('Sync v3 sign up and CWTS', () => {
+  test.beforeEach(async ({ target }) => {
+    //Sync tests run a little slower and flake
+    test.slow();
+    syncBrowserPages = await newPagesForSync(target);
+    const { login } = syncBrowserPages;
+    email = login.createEmail('sync{id}');
+  });
+
+  test.afterEach(async () => {
+    await syncBrowserPages.browser?.close();
+  });
+
+  test('verify with signup code and CWTS', async ({ target }) => {
+    const { login, page, signinTokenCode, connectAnotherDevice } =
+      syncBrowserPages;
+    const query = new URLSearchParams({
+      forceUA: uaStrings['desktop_firefox_71'],
+    });
+    const eventDetailStatus = createCustomEventDetail(
+      FirefoxCommand.FxAStatus,
+      {
+        signedInUser: null,
+        capabilities: {
+          choose_what_to_sync: true,
+          multiService: true,
+          engines: ['history'],
+        },
+      }
+    );
+    const eventDetailLinkAccount = createCustomEventDetail(
+      FirefoxCommand.LinkAccount,
+      {
+        ok: true,
+      }
+    );
+    await page.goto(
+      `${
+        target.contentServerUrl
+      }?context=fx_desktop_v3&service=sync&automatedBrowser=true&${query.toString()}`
+    );
+    await login.respondToWebChannelMessage(eventDetailStatus);
+    await login.respondToWebChannelMessage(eventDetailLinkAccount);
+    await login.checkWebChannelMessage('fxaccounts:fxa_status');
+    await login.setEmail(email);
+    await signinTokenCode.clickSubmitButton();
+    await login.setPassword(password);
+    await login.confirmPassword(password);
+    await login.setAge('21');
+
+    // The CWTS form is on the same signup page
+    await login.waitForCWTSEngineHeader();
+    expect(await login.isCWTSEngineBookmarks()).toBe(true);
+    expect(await login.isCWTSEngineHistory()).toBe(true);
+    expect(await login.isCWTSEnginePassword()).toBe(true);
```

Comment:
Related to previous suggestion above about renaming this method to be plural for consistency:
```suggestion
    expect(await login.isCWTSEnginePasswords()).toBe(true);
```